### PR TITLE
Add BackendConfig specification for customizable health checks [WIP]

### PR DESCRIPTION
**Pull request by rramkumar1**
Addresses #42.

This PR by itself simply defines the schema for customizable health checks and adds some helpers. There is still a bit of plumbing that needs to be done to make this all work.

### CODE REVIEW

File: pkg/backendconfig/validation.go [hunk 1]
```diff
@@ -104,3 +118,54 @@ func validateSessionAffinity(kubeClient kubernetes.Interface, beConfig *backendc
 
 	return nil
 }
+
+// TODO(rramkumar): When we migrate to kubebuilder this sort of validation will be done in the Go type definition.
+func validateHealthCheck(kubeClient kubernetes.Interface, beConfig *backendconfigv1beta1.BackendConfig) error {
```
Comment (julianvmodesto) [inline pkg/backendconfig/validation.go]: Add TestValidHealthCheck?

File: pkg/backendconfig/validation.go [hunk 2]
```diff
@@ -104,3 +118,54 @@ func validateSessionAffinity(kubeClient kubernetes.Interface, beConfig *backendc
...
+		}
+	}
+
+	checkInterval := beConfig.Spec.HealthCheck.CheckIntervalSec
+	if checkInterval != nil {
+		if *checkInterval <= 0 || *checkInterval > 300 {
+			return fmt.Errorf("health check interval must be greater than 0 and less than or equal to 300, got %d", *checkInterval)
+		}
+	}
+
```
Comment (julianvmodesto) [inline pkg/backendconfig/validation.go]: check for timeout > check interval?

```go
if *timeout > *checkInterval {
  return fmt.Errorf("health check timeout %d cannot be greater than check interval %d", *timeout, *checkInterval)
}
```

**Comment by julianvmodesto** [comment]
Can this PR proceed? Not sure if it was not targeted for this past release (or maybe still isn't)

**Comment by rramkumar1** [comment]
@julianvmodesto Yes this is still on the table for our v1.6 release. We are still releasing patches for v1.5 to get some bug fixes out.

**Comment by julianvmodesto** [comment]
Gotcha, appreciate it!

**Comment by rramkumar1** [comment]
@julianvmodesto There has been a slight change of plans regarding how we will be surfacing customizable health checks. 

As a result, I am going to close this PR. Apologies for this but please know that this support is a priority.

**Comment by dhan** [comment]
@rramkumar1 in that case is there somewhere else that we can track this feature?

**Comment by rramkumar1** [comment]
@dhan You can continue to track #42.

**Pull request closed without merging**